components/sidebar/sidebar_channel/sidebar_group_channel/index.ts

```diff
@@ -2,21 +2,60 @@
 // See LICENSE.txt for license information.
 
 import {connect} from 'react-redux';
-import {bindActionCreators, Dispatch} from 'redux';
+import {bindActionCreators, Dispatch, ActionCreatorsMapObject} from 'redux';
 
+import {savePreferences} from 'mattermost-redux/actions/preferences';
+import {ActionFunc} from 'mattermost-redux/types/actions';
+import {PreferenceType} from 'mattermost-redux/types/preferences';
 import {GlobalState} from 'mattermost-redux/types/store';
-import {GenericAction} from 'mattermost-redux/types/actions';
+import {Channel} from 'mattermost-redux/types/channels';
+import {UserProfile} from 'mattermost-redux/types/users';
+import {getCurrentChannelId, getRedirectChannelNameForTeam} from 'mattermost-redux/selectors/entities/channels';
+import {getCurrentTeam} from 'mattermost-redux/selectors/entities/teams';
+import {getUserIdsInChannels, getCurrentUser} from 'mattermost-redux/selectors/entities/users';
 
 import SidebarGroupChannel from './sidebar_group_channel';
 
-function mapStateToProps(state: GlobalState) {
+type OwnProps = {
+    channel: Channel;
+}
+
+function mapStateToProps(state: GlobalState, ownProps: OwnProps) {
+    const currentUser = getCurrentUser(state);
+    const currentTeam = getCurrentTeam(state);
+    const redirectChannel = getRedirectChannelNameForTeam(state, currentTeam.id);
+    const currentChannelId = getCurrentChannelId(state);
+    const active = ownProps.channel.id === currentChannelId;
+
+    const memberIds = getUserIdsInChannels(state);
+
+    let membersCount = 0;
+    if (memberIds && memberIds[ownProps.channel.id]) {
+        const groupMemberIds: Set<string> = memberIds[ownProps.channel.id] as unknown as Set<string>;
```

Comment:
`as unknown as Set<string>`?

Reply:
We have some bad typings here, `getUserIdsInChannels` is actually returning a dictionary of type `<string, Set<string>>`, but our type definitions are saying it's a `RelationOneToMany<Channel, UserProfile>`. We need to fix these type definitions, but that's a bigger undertaking outside the scope of this PR.

Reply:
Ditto here. Can you create a ticket for that?

Reply:
https://mattermost.atlassian.net/browse/MM-22356